### ISSUE TITLE
Updating `DoSrvPrivateReverseLineFeed` with a direct call to `ScrollRegion`?

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#2702]
Would it not be worthwhile updating `DoSrvPrivateReverseLineFeed` at the same time, assuming it's just a matter of replacing the `ScrollConsoleScreenBufferWImpl` call with a direct call to `ScrollRegion`?

As for the testing, I've added a whole bunch of scrolling tests in PR #2505 which I think could easily be adapted to use 256 colors once this is merged. Although that PR is going to need to be rebased and updated first.

_Originally posted by @j4james in https://github.com/microsoft/terminal/pull/2668#issuecomment-528646590_

<hr> 

We should also make sure to add some tests for this case as well. I'll do it as a separate PR to make sure we get each one in correctly.
